cluster_experiments/experiment_analysis.py

```diff
@@ -539,3 +539,108 @@ def from_config(cls, config):
+
+    Usage:
+
+    ```python
+    from cluster_experiments.experiment_analysis import MLMExperimentAnalysis
+    import pandas as pd
+
+    df = pd.DataFrame({
+        'x': [1, 2, 3, 0, 0, 1],
+        'treatment': ["A"] * 3 + ["B"] * 3,
+        'cluster': [1] * 6,
+    })
+
+    MLMExperimentAnalysis(
+        cluster_cols=['cluster'],
+        target_col='x',
+    ).get_pvalue(df)
+    ```
+    """
+
+    def __init__(
+        self,
+        cluster_cols: List[str],
+        target_col: str = "target",
+        treatment_col: str = "treatment",
+        treatment: str = "B",
+        covariates: Optional[List[str]] = None,
+    ):
+        super().__init__(
+            target_col=target_col,
+            treatment_col=treatment_col,
+            cluster_cols=cluster_cols,
+            treatment=treatment,
+            covariates=covariates,
+        )
+        # the specification with C() ensures consistent ordering of treatment and non-treatment in the model
+        self.mlm_treatment_spec = (
+            f"C({self.treatment_col}, Treatment(reference='{self.treatment}'))"
+        )
+        self.mlm_treatment_label = None
+        self.formula = f"{self.target_col} ~ {self.mlm_treatment_spec}"
+        if self.covariates:
+            self.formula = f"{self.formula} + {' + '.join(self.covariates)}"
+
+        self.re_formula = None
+        self.vc_formula = None
+
+    def fit_mlm(self, df: pd.DataFrame) -> sm.MixedLM:
+        """Returns the fitted MLM model"""
+        self.mlm_treatment_label = (
+            f"{self.mlm_treatment_spec}[T.{self._get_non_treatment_label(df)}]"
+        )
+        return sm.MixedLM.from_formula(
+            formula=self.formula,
+            data=df,
+            groups=self._get_cluster_column(df),
+            re_formula=self.re_formula,
+            vc_formula=self.vc_formula,
+        ).fit()
+
+    def _get_non_treatment_label(self, df):
+        return (set(df[self.treatment_col]) - set([self.treatment])).pop()
+
+    def analysis_pvalue(self, df: pd.DataFrame, verbose: bool = False) -> float:
+        """Returns the p-value of the analysis
+        Arguments:
+            df: dataframe containing the data to analyze
+            verbose (Optional): bool, prints the regression summary if True
+        """
+        results_mlm = self.fit_mlm(df)
+        if verbose:
+            print(results_mlm.summary())
+        return results_mlm.pvalues[self.mlm_treatment_label]
+
+    def analysis_point_estimate(self, df: pd.DataFrame, verbose: bool = False) -> float:
+        """Returns the point estimate of the analysis
+        Arguments:
+            df: dataframe containing the data to analyze
+            verbose (Optional): bool, prints the regression summary if True
+        """
+        results_mlm = self.fit_mlm(df)
+        # negative sign, as we use the treatment as reference (instead of the baseline)
+        return -results_mlm.params[self.mlm_treatment_label]
```

Comment:
for consistency with the other classes in the library, could we return the usual
```python
        return results_mlm.params[self.mlm_treatment_label]
```
and you change the sign in your code?

Reply:
Hmmm my thinking was that applying the sign here is precisely so that the output of the method is aligned with the output of the same method in the other classes... I feel that it isn't optimal for the user to always have to think about reversing the sign when using this class. What do you think? Do you have another argument to convince me here?^^

Reply:
so, does `-results_mlm.params[self.mlm_treatment_label]` give an estimate of the ATE? if I run, with the same data a ClusteredOls and a MixedLM, do I get similar results of get_point_estimate? or is the sign changed?

Reply:
Correct, with the negative sign the point estimate result is the same as for ClusteredOLS. When I tried ClusteredOLS by the way, I found that using the df that's in the docstring it doesn't work: The reason is that in the params of the fitted ols model, the code is looking for the label `"treatment"`, but actually it's `"treatment[T.B]"`: This is precisely the case I wanted to avoid in my code and where the negative sign results from (see the comment I left [here](https://github.com/david26694/cluster-experiments/blob/7eb983ef08de088caf9ba25c3a71110582035946/cluster_experiments/experiment_analysis.py#L589)). I'm not exactly sure how `sm.OLS` behaves, but in the case of `sm.MixedLM`, the assignment of which of the labels in the `"treatment"` column is considered to be the "baseline" and which is the "treated" is (I believe) simply based on which of the two labels ("A" or "B" in the case of the test df) appears first. So I explicitly made it deterministic to always be the "treated" ("B" in our case) first (the only reason being that we already have this label in `self.treatment`, whereas we don't have any information on the "baseline label"). In order to then convert to ATE, I have to take the negative sign.
Perhaps it would be good to do something similar in the `ClusteredOLSAnalysis` class?